src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,134 @@
+import "./Carousel.css";
+import React, { ReactElement, useEffect, useRef, useState } from "react";
+import { LEFT, RIGHT, GAP } from "../../constants/constant";
+import CarouselButton from "./CarouselButton";
+
+const CLEAREANCE_RIGHT: number = 300;
+
+interface IButtonVisiblity {
+  left: boolean;
+  right: boolean;
+}
+
+const Carousel = (): ReactElement => {
+  const carouselItem = useRef<HTMLDivElement>(null);
+  const carouselRef = useRef<HTMLDivElement>(null);
+  const carouselContainerRef = useRef<HTMLDivElement>(null);
+
+  const [btnVisible, setbtnVisble] = useState<IButtonVisiblity>({
+    left: false,
+    right: true,
+  });
+
+  const { left, right } = btnVisible;
+
+  const checkVisiblity = (direction: string) => {
```

Comment:
this function is not only checking visibility it is changing state also. Please find a better name for this